### ISSUE TITLE
User Guide

### ISSUE THREAD

**Issue by LanceGundersen** [LanceGundersen/navfit2020#15]
Create Wiki for how to use the application.
- [ ] Description of app
- [ ] How to Install and run app
- [ ] How to setup defaults
- [ ] How to add, edit and delete Sailors
- [ ] How to add, edit and delete Evals
- [ ] How to export entire database (not implemented yet)
- [ ] How to export single Sailor (not implemented yet)
- [ ] How to archive single Sailor (not implemented yet)

**Comment by JohnHaTrick** [comment]
Hi Lance,

I stumbled across this project and would love to help if I can. Documentation seems like a good place to pitch in and simultaneously get oriented.

I'm quite familiar with NAVFIT98, linux, and am a beginner in Java. I'm fairly experienced with C++ and Python, though.

Do you have a high-level game-plan for this project?

V/r, John

**Comment by LanceGundersen** [comment]
@JohnHaTrick, 

I was considering stopping development as I think it won't meet the needs of anyone besides the reserve community. I also got feedback that it likely wouldn't ever be authorized to be used in any official capacity either. 

That said, it is all javascript based so pretty simple to make and I was having fun but it is just me so far really. Off the cuff, I think we need a development plan to get to an MVP and under that, there are some bigger details to be worked such as actually filling out a PDF and some work on the math.

If you're seriously interested we can get a meeting together and come up with a game plan! 🎉 

Lance

**Comment by JohnHaTrick** [comment]
Gotcha Lance,

Did you hear that big Navy spent big bucks paying someone to develop a replacement for NAVFIT? But when it was finished the Navy had messed up the requirements or something (hearsay) and they ended up scrapping the whole project.

I don't want to over-promise because my time is pretty saturated, but I'm definitely interested.

I don't think the authorization of the program is necessarily a big hang-up. I don't think anyone would be breaking the rules by adopting it at the grassroots level.

**Comment by LanceGundersen** [comment]
Oh really? That would fill in the gap as to `why` nothing has been done yet 😢 

I am saturated too but generally open up over the winter months but another set of eyes would probably unblock me on the little things that are bogging me down though.

I think if we stay focused on the reservists and enlisted evals first for the MVP it would streamline the requirements a bit.

**Comment by JohnHaTrick** [comment]
I just emailed the Captain who is overseeing the Performance Evaluation Transformation (PET), to let him know your project exists.

Maybe something cool will happen.

Btw, I'm a LCDR in HR (1205). My current duty as a reservist is at Office of Naval Research on a taxonomy for AI. My fulltime gig is working on my PhD research on contingency planning for self-driving cars.

**Comment by mpyne-navy** [comment]
@JohnHaTrick I'm also an HR (1200 though), working at OPNAV N1. Didn't know there were that many technical minds out there in our community, besides the NPS OA grads that is.

There was a separate effort by some HR officers to build an R-based model to help optimize the order in which NPC started to cut orders (see https://www.navy.mil/DesktopModules/ArticleCS/Print.aspx?PortalId=1&ModuleId=763&Article=2287378) so things can happen sometimes. But that only worked because this team was able to run R on their NMCI seats, as soon as you start talking about dedicated IT systems everything freezes up again :(

**Comment by JohnHaTrick** [comment]
Hello Michael! That's a pretty cool story, and I think should there should be more like it.

I'm currently trying to think about what my next steps should be in HR. I want to find something that utilizes my technical training, but having trouble so far. I'm thinking maybe at DIU?

**Comment by mpyne-navy** [comment]
@JohnHaTrick I'm sure DIU could use your talent but I don't know that they get *technical* very much either, I believe they are more focused on acquisition excellence to onboard private sector innovation even more efficiently, could be wrong though. There is a team trying to bootstrap the Air Force's successful DevOps work for DON usage, called [Black Pearl](https://www.linkedin.com/company/don-black-pearl/?miniCompanyUrn=urn%3Ali%3Afs_miniCompany%3A69243939) based off of USAF's [Platform One](https://software.af.mil/dsop/services/), but this is very much in its infancy. Though @LanceGundersen you would probably be interested in Black Pearl as well if it can get going!

Here in OPNAV N1, where I work, there's a lot of room for HRs and our leadership are always trying to find more, but the work has not been very technical yet, though that might change if we can get a nucleus of the right talent.

There's also efforts like Lt. Sean Lavelle's at NAVAIR (I think) and Cmdr. Jarrod Groves at CNAP (the "FRAG", Force Readiness Analytics Group). The former is more "Javascript on a single web page for TACAIDs that cyber can't prevent from shipping" and the latter is more "data wrangling and dashboards", neither are really C++/Java or writing backend programs, but at least they're technical.

I'd like to say NAVWAR or its affiliates but as far as I can tell the government side is mostly focused on oversight and management of technical contractors instead of getting their own hands dirty, which is a shame. Hard to stay technically proficient when you spend a career in management!

**Comment by SemperRuminare** [comment]
Hey! This seems cool. I tried to get this to run on my local machine but I'm having issues.  (navfit2020/node_modules/electron/dist/electron: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory)

Would love to get this thing running, not sure why it's not playing nice with Ubuntu. Let me know if there's an invite to this meeting!

Also, we run Marine Coders and we can spotlight your project there (marinecoders.github.io). No Navy Coders yet, but Black Pearl is pretty close. Soon!

**Comment by LanceGundersen** [comment]
@SemperRuminare try the dev branch I haven't updated master in quite a while since I'm still pre-alpha. It's hot and loose at this point. 

I want to get the the majority of the eval exporting properly in the next week or two before setting up a meeting for milestones and a roadmap.

**Comment by mpyne-navy** [comment]
@SemperRuminare try installing the `libnss3` package on your Ubuntu first (e.g. `sudo apt install libnss3`). This package is a cryptographic handling package provided by Mozilla.org and used by lots of Web-handling libraries and applications including Firefox, Chrome (on Linux), and Node.js. But it's typically provided by the OS, I don't think it's available as an NPM package.

**Comment by FelicianoTech** [comment]
> I was considering stopping development as I think it won't meet the needs of anyone besides the reserve community.

@LanceGundersen  Hmm. It looks like you changed your mind? You seem to have been doing well so far and should continue. For whatever reason if you did want to stop, I'd love to continue it for you.

> Also, we run Marine Coders

@SemperRuminare Nice! A shipmate and I run https://www.TheReserveForce.com a resource for Navy reservists though we're actively trying to add content for other branches.

---

I'd like to add that in addition to a wiki, a static site built with Jekyll (like Marine Coders) or Gotham (like The Reserve Force) can be an option as well. GitHub Pages provides free hosting for static sites and I would also be willing to offer up hosting if you went that route.

**Comment by LanceGundersen** [comment]
@felicianotech Yes, I am continuing to develop due to getting some feedback that there were barriers (i.e. the OSs security warnings) blocking early feedback loops. Now that the app has been submitted to the Windows store it has 15 downloads to date.
